components/server/ee/src/workspace/gitpod-server-impl.ts

```diff
@@ -1904,18 +1904,22 @@ export class GitpodServerEEImpl extends GitpodServerImpl {
         }
     }
 
-    async findStripeCustomerIdForTeam(ctx: TraceContext, teamId: string): Promise<string | undefined> {
-        const user = this.checkAndBlockUser("findStripeCustomerIdForTeam");
+    async findStripeSubscriptionIdForTeam(ctx: TraceContext, teamId: string): Promise<string | undefined> {
+        const user = this.checkAndBlockUser("findStripeSubscriptionIdForTeam");
         await this.ensureIsUsageBasedFeatureFlagEnabled(user);
         await this.guardTeamOperation(teamId, "update");
         try {
             const customer = await this.stripeService.findCustomerByTeamId(teamId);
-            return customer?.id || undefined;
+            if (!customer?.id) {
+                return undefined;
```

Comment:
As in another PR, I think this should be returning NotFound and the client handling not found as No subscription exists.

Reply:
Until now, I think we've had our API return `something | undefined` for such `findSomething` methods.

We could change this pattern, but then maybe it would make sense to change all these methods?

Reply:
Up to you. I'd mostly suggest we do this opportunistically on apis we work on as part of features.